test/unit/test-video-analytics-percentage-tracker.js

```diff
@@ -148,6 +148,30 @@ describes.fakeWin(
         });
       });
 
+      // TODO(#25954): This test is a bit specific and odd, but replicates
+      // what we see in prod. Possibly remove when root cause for video with
+      // duration => no duration is found.
+      [0, NaN, -1, undefined, null].forEach(invalidDuration => {
+        it(`aborts if duration is ${invalidDuration} after initially valid`, () => {
+          const {video} = mockEntry;
+          const durationStub = env.sandbox.stub(video, 'getDuration');
+          durationStub.onFirstCall().returns(1000 /* valid duration */);
+          durationStub.returns(invalidDuration);
+
+          const {element} = video;
+
+          setPlayingState(mockEntry, PlayingStates.PLAYING_MANUAL);
+          tracker.start();
+
+          dispatchLoadedMetadata(element);
+
+          expect(mockTimer.delay).to.not.have.been.calledWith(
```

Comment:
This should check that this is called with the constant frequency for `paused` state, no?

Reply:
Not sure I understand fully. This just ensures the `delay()` method  is never passed the "invalid" values that could cause the infinite loop. Is there something else you are looking for?

Reply:
The invalid values are never passed through in any case. I mean that it should check the call on the conditional:

```js
timer.delay(calculateAgain, PERCENTAGE_FREQUENCY_WHEN_PAUSED_MS);
```

Reply:
Done.